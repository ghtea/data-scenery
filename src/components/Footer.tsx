import React, { useCallback, useEffect } from "react";
import history from 'historyApp';
import { useLocation } from "react-router-dom";
import { FormattedMessage } from 'react-intl';

import {useSelector, useDispatch} from "react-redux";
import {StateRoot} from 'store/reducers';
import * as actionsRoot from "store/actions";

import styles from './Footer.module.scss';

//import IconLogo from 'svgs/others/IconLogo';

type PropsFooter = {};

function Footer({}: PropsFooter) {

    const dispatch = useDispatch();
    const location = useLocation();
    
    const showingFooter = useSelector((state: StateRoot) => state['status']['showing']['footer']);
  
  
  return (
    <footer className={`${styles['root']} showing----${showingFooter}`}>
        <div> â“’ 2016. nextwing.me All Rights Reserved. </div> 
    </footer>
  );
}

Footer.defaultProps = {};

export default Footer;
