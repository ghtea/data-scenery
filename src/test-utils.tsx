import React from 'react';
import { render as renderRtl, RenderOptions } from '@testing-library/react';
// import '@testing-library/dom';

import { IntlProvider } from 'react-intl';
import translationEn from 'language/translation/en.json';

// action 하나로 합쳐볼까?!!!
// import * as actionsRoot from 'store/actions'; 

import { Router} from "react-router-dom";  // BrowserRouter
import history from 'historyApp';

import {Provider, useSelector, useDispatch} from 'react-redux' 
import store from './store';

import { CookiesProvider } from 'react-cookie';

// import '@testing-library/jest-dom/extend-expect'; // 지금으로선 아마 각각의 파일에 추가해야 할듯 
export {screen, fireEvent} from '@testing-library/react';
export {waitFor} from '@testing-library/dom'
//import userEvent from '@testing-library/user-event';


export const beforeAllDefault = () => {
    beforeAll(()=>{
        jest.spyOn(console, 'log').mockImplementation(jest.fn());
        jest.spyOn(console, 'group').mockImplementation(jest.fn());
    });
}


// wrote automatically from IDE tooltip
type Ui = React.ReactElement<any, string | ((props: any) => React.ReactElement<any, string | any | (new (props: any) => React.Component<any, any, any>)> | null) | (new (props: any) => React.Component<any, any, any>)>;

type Option = Pick<RenderOptions<typeof import("@testing-library/dom/types/queries")>, "container" | "baseElement" | "hydrate" | "wrapper"> | undefined;

export const render = (ui: Ui, option: Option = undefined) =>{

    const wrapper:React.FunctionComponent<{}> = ({ children }) => {

        return (
            <Router history={history}>
            <CookiesProvider>
            <Provider store={store}>
            <IntlProvider locale={'en'} messages={translationEn} >
                {children}
            </IntlProvider>
            </Provider>
            </CookiesProvider>
            </Router>
        )
    }

    return renderRtl(ui, { wrapper: wrapper, ...option })
}



/*

import { 
    render, beforeAllDefault, 
    screen, fireEvent, waitFor
} from 'test-utils'; 
import '@testing-library/jest-dom/extend-expect';


*/