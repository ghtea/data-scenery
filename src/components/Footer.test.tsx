import React from 'react';
import { render} from 'test-utils';
// import '@testing-library/jest-dom/extend-expect';    // 이거 필요없는 듯?
import {screen, fireEvent} from '@testing-library/react';
//import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom/extend-expect';
// import * from '@testing-library/dom';

import translationEn from 'language/translation/en.json';
import Footer from './Footer';
// import Modal from './Modal';

// import { StringLiteral } from 'typescript';


describe('<Footer />', () => {

    it('matches snapshot', () => {
        const {container} = render(<Footer />);
        expect(container).toMatchSnapshot();
    });

    
});

