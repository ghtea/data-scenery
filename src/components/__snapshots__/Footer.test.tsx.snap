// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> matches snapshot 1`] = `
<div>
  <footer
    class="root showing----false"
  >
    <div>
       â“’ 2016. nextwing.me All Rights Reserved. 
    </div>
  </footer>
</div>
`;
